app/src/org/commcare/preferences/HiddenPreferences.java

```diff
@@ -631,10 +631,23 @@ public static boolean isBackgroundSyncEnabled() {
         return DeveloperPreferences.doesPropertyMatch(ENABLE_BACKGROUND_SYNC, PrefValues.NO, PrefValues.YES);
     }
 
-    public static void setInterruptedFormIndex(String serializedFormIndex) {
+    public static void setInterruptedFormIndex(int ssdId, String serializedFormIndex) {
         String currentUserId = CommCareApplication.instance().getCurrentUserId();
         CommCareApplication.instance().getCurrentApp().getAppPreferences().edit()
-                .putString(INTERRUPTED_FORM_INDEX + currentUserId,  serializedFormIndex)
+                .putString(INTERRUPTED_FORM_INDEX + currentUserId, ssdId + " " + serializedFormIndex)
```

Comment:
think it's better to have `ssdId` as part of key rather than value. It will simplify code while retrieving the form index as well.

Reply:
That's a good point.